docs/devguide/building/index.rst

```diff
@@ -527,3 +529,171 @@ Users of OpenDDS are strongly encouraged to select one of the two options listed
+
+  - See https://www.dre.vanderbilt.edu/~schmidt/DOC_ROOT/ACE/ACE-INSTALL.html for how to manually build ACE/TAO.
+
+The following is an example of cross-compiling OpenDDS for Android on Linux using CMake.
+It assumes the NDK has been downloaded and the location is in an environment variables called ``NDK`` and the downloaded ACE/TAO version matches the version being used by host tools.
+
+.. code-block:: shell
+
+  # Build Host Tools
+  cmake -B build-host -DOPENDDS_JUST_BUILD_HOST_TOOLS=TRUE
+  cmake --build build-host -- -j 4
+  export HOST_ROOT=$(realpath build-host/ace_tao)
+
+  # Build ACE/TAO for Android
+  curl -L https://github.com/DOCGroup/ACE_TAO/releases/download/ACE+TAO-7_1_0/ACE+TAO-7.1.0.tar.gz \
+    --output ACE+TAO.tar.gz
+  tar xzf ACE+TAO.tar.gz
+  export ACE_ROOT=$(realpath ACE_wrappers)
+  export TAO_ROOT=$ACE_ROOT/TAO
+  export MPC_ROOT=$ACE_ROOT/MPC
+  echo 'no_cxx11=0' > "$ACE_ROOT/bin/MakeProjectCreator/config/default.features"
+  cat << EOF > "$ACE_ROOT/include/makeinclude/platform_macros.GNU"
+  optimize = 0
+  android_abi := armeabi-v7a
+  android_api := 29
+  android_ndk := \$(NDK)
+  TAO_IDL := \$(HOST_ROOT)/bin/tao_idl
+  TAO_IDLFLAGS += -g \$(HOST_ROOT)/ace_gperf
+  TAO_IDL_DEP := \$(HOST_ROOT)/bin/tao_idl
+  include \$(ACE_ROOT)/include/makeinclude/platform_android.GNU
+  EOF
+  cat << EOF > "$ACE_ROOT/ace/config.h"
+  #define ACE_DISABLE_MKTEMP
+  #define ACE_LACKS_READDIR_R
+  #define ACE_LACKS_TEMPNAM
+  #include "config-android.h"
+  EOF
+  cp ACE_TAO_for_OpenDDS.mwc $ACE_ROOT
+  (cd $ACE_ROOT && bin/mwc.pl -type gnuace ACE_TAO_for_OpenDDS.mwc && make -j 4)
+
+  # Build OpenDDS for Android
+  cmake -B build-target \
+    -DBUILD_SHARED_LIBS=TRUE \
+    -DANDROID_ABI=armeabi-v7a -DANDROID_PLATFORM=android-29 \
+    --toolchain $NDK/build/cmake/android.toolchain.cmake \
+    -DOPENDDS_ACE_TAO_HOST_TOOLS=$(realpath build-host/ace_tao) \
+    -DOPENDDS_HOST_TOOLS=$(realpath build-host) \
+    -DOPENDDS_ACE=$ACE_ROOT
+  cmake --build build-target -- -j 4
+
+Installation
+============
+
+Once built, OpenDDS can be installed using `cmake --install <https://cmake.org/cmake/help/latest/manual/cmake.1.html#install-a-project>`__.
+Currently ACE/TAO has to be installed separately and this is only possible with GNU Make.
+
+.. _cmake-running-tests:
+
+Running Tests
+=============
+
+Tests (:cmake:var:`OPENDDS_BUILD_TESTS`) and examples (:cmake:var:`OPENDDS_BUILD_EXAMPLES`) can be run using the ``test`` target (``cmake --build build -t test``) or directly using `ctest <https://cmake.org/cmake/help/latest/manual/ctest.1.html>`__.
+
+Other Known Limitations
+=======================
+
+- The following features are planned, but not implemented yet:
+
+  - The ability to use MPC for building user applications.
+  - Safety profile
+  - Java Mapping
+
+- CMake-build OpenDDS libraries and executables will currently be ignored by :ref:`find_package(OpenDDS COMPONENTS ...) <cmake-components>`.
+  Passing feature should work.
```

Comment:
I'm not sure what this line means

Reply:
Besides it should saying "features", I might have worded it poorly. Here is the full explanation:

In an MPC-built OpenDDS [features, libraries, and executables can be passed to find_package to require that they are enabled/disable/available](https://opendds.readthedocs.io/en/master/devguide/building/cmake.html#components). For example `find_package(OPENDDS REQUIRED safety_profile=OFF OpenDDS::Security OpenDDS::RtpsRelay)` will require that safety profile is disabled and the relay and security library can be used if OpenDDS is found. This also works within a CMake-build OpenDDS.

Once the CMake-built OpenDDS is installed though a different target direct importing mechanism has to be used, which is decided [here](https://github.com/iguessthislldo/OpenDDS/blob/d1cdb48a76bcefca6702fccda4332b791998477a/cmake/OpenDDSConfig.cmake#L222-L234). I haven't worked out exactly how the installed mechanism interacts with `find_package` yet. Theoretically it shouldn't be too hard, just check if the target exists, but I just haven't gotten around to it. ACE/TAO targets always work because they're always imported indirectly, which I should probably mention. Features also always work because these are just variables and don't really have much to do with the target importing mechanisms.

Reply:
I have implemented the check, so this note is no longer necessary.